### PR TITLE
Have real workflow name as part of resource name

### PR DESCRIPTION
**Pull request by anniezhang77**
risk: low

### CODE REVIEW

**Review by damianxu88 — APPROVED** [review]
LGTM

**Review by schowsf — APPROVED** [review]

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala [hunk 1]
```diff
@@ -232,6 +232,7 @@ object Ec2Resource {
     instanceProfile: String,
     securityGroups: Option[Seq[String]],
     tags: Option[Seq[AwsTag]],
+    workflowName: Option[String],
```
Comment (realstraw) [inline orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala:235]: let's just call it name

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala [hunk 2]
```diff
@@ -240,7 +241,8 @@ object Ec2Resource {
   def decode(conf: ResourceIO.Conf): JsResult[Ec2Resource] = conf.resourceSpec
     .validate[Spec]
     .map { spec =>
-      Ec2Resource.apply(s"${conf.workflowId}_rsc-${conf.resourceId}_${conf.instanceId}", spec)
+      val name = s"${conf.workflowId}_rsc-${conf.resourceId}_${spec.workflowName.fold("")(s => s"${s}_")}${conf.instanceId}"
```
Comment (realstraw) [inline orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala:244]: let's use auto generated name if `name` is not configured, and just simply use `name` if it's supplied

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/EmrResource.scala [hunk 3]
```diff
@@ -235,14 +235,16 @@ object EmrResource {
     tags: Option[Seq[AwsTag]],
     bootstrapActions: Option[Seq[BootstrapAction]],
     configurations: Option[Seq[ConfigurationSpec]],
+    workflowName: Option[String],
```
Comment (realstraw) [inline orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/EmrResource.scala:238]: same here

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala (hunk 2 above)
Comment (anniezhang77) [reply orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala:244]: ok

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala (hunk 1 above)
Comment (anniezhang77) [reply orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala]: done

**Pull request merged**